Modules/_ctypes/stgdict.c

```diff
@@ -411,231 +372,63 @@ PyCStructUnionType_update_stginfo(PyObject *type, PyObject *fields, int isStruct
-            const char *fieldfmt = info->format ? info->format : "B";
-            const char *fieldname = PyUnicode_AsUTF8(name);
-            char *ptr;
-            Py_ssize_t len;
-            char *buf;
-            Py_ssize_t last_size = size;
-            Py_ssize_t padding;
-
-            if (fieldname == NULL)
-            {
-                Py_DECREF(pair);
-                return -1;
-            }
 
-            /* construct the field now, as `prop->offset` is `offset` with
-               corrected alignment */
-            prop = PyCField_FromDesc(st, desc, i,
-                                   &field_size, bitsize, &bitofs,
-                                   &size, &offset, &align,
-                                   pack, big_endian, layout_mode);
-            if (prop == NULL) {
-                Py_DECREF(pair);
-                return -1;
-            }
+        assert(prop);
 
-            /* number of bytes between the end of the last field and the start
-               of this one */
-            padding = ((CFieldObject *)prop)->offset - last_size;
-
-            if (padding > 0) {
-                ptr = stginfo->format;
-                stginfo->format = _ctypes_alloc_format_padding(ptr, padding);
-                PyMem_Free(ptr);
-                if (stginfo->format == NULL) {
-                    Py_DECREF(pair);
-                    Py_DECREF(prop);
-                    return -1;
-                }
-            }
-
-            len = strlen(fieldname) + strlen(fieldfmt);
-
-            buf = PyMem_Malloc(len + 2 + 1);
-            if (buf == NULL) {
-                Py_DECREF(pair);
-                Py_DECREF(prop);
-                PyErr_NoMemory();
-                return -1;
-            }
-            sprintf(buf, "%s:%s:", fieldfmt, fieldname);
-
-            ptr = stginfo->format;
-            if (info->shape != NULL) {
-                stginfo->format = _ctypes_alloc_format_string_with_shape(
-                    info->ndim, info->shape, stginfo->format, buf);
-            } else {
-                stginfo->format = _ctypes_alloc_format_string(stginfo->format, buf);
-            }
-            PyMem_Free(ptr);
-            PyMem_Free(buf);
-
-            if (stginfo->format == NULL) {
-                Py_DECREF(pair);
-                Py_DECREF(prop);
-                return -1;
-            }
-        } else /* union */ {
-            field_size = 0;
-            size = 0;
-            bitofs = 0;
-            offset = 0;
-            align = 0;
-            prop = PyCField_FromDesc(st, desc, i,
-                                   &field_size, bitsize, &bitofs,
-                                   &size, &offset, &align,
-                                   pack, big_endian, layout_mode);
-            if (prop == NULL) {
-                Py_DECREF(pair);
-                return -1;
-            }
-            union_size = max(size, union_size);
+        if (-1 == PyObject_SetAttr(type, prop->name, prop_obj)) {
```

Comment:
```suggestion
        if (PyObject_SetAttr(type, prop->name, prop_obj) < 0) {
```

To be consistent.

Reply:
Why not. That's a line I didn't touch, BTW; unfortunately GitHub's diff algorithm doesn't realize that.

Reply:
Oh ok. Then fine!